### PR TITLE
Revamps Lavaland - (Re)Building a Life Edition

### PR DESCRIPTION
**Pull request by Son-of-Space**
## About The Pull Request

PICTURE CATALOG STARTS HERE:

<details>
<summary> Mining base full</summary>

image

</details>

<details>
<summary> Public mining full</summary>

image

</details>

<details>
<summary> The public receiving area </summary>

image

</details>

<details>
<summary> Tcomms and the ruined prisoner transfer bridge </summary>

image

</details>

<details>
<summary> DonkCo sponsored commissary and public medical bay </summary>

image

</details>

<details>
<summary> Public Mining Maintenance </summary>

image

</details>

<details>
<summary> Public Mining Exit and Dorms </summary>

image

</details>

<details>
<summary> Careful, the exit is a little precarious.</summary>

image

</details>

<details>
<summary> Public diner</summary>

image

</details>

<details>
<summary> Public processing</summary>

image

</details>

<details>
<summary> Public bathroom</summary>

image

</details>

<details>
<summary> Public mechbay</summary>

image

</details>

<details>
<summary> Mining dock and dorms</summary>

image

</details>

<details>
<summary> Mining prep area</summary>

image

</details>

<details>
<summary> Gulag station full</summary>

image

</details>

<details>
<summary> Gulag island 1</summary>

image

</details>

<details>
<summary> Gulag island 2 1/2</summary>

image

</details>

<details>
<summary> Gulag island 2 2/2</summary>

image

</details>

<details>
<summary> Gulag islands attachment</summary>

image

</details>

<details>
<summary> Gulag commons</summary>

image

</details>

<details>
<summary> Gulag sec hallway</summary>

image

</details>

<details>
<summary> Gulag sec outpost</summary>

image

</details>

<details>
<summary> Gulag living area</summary>

image

</details>

<details>
<summary> Gulag "storage room"</summary>

image

</details>

<details>
<summary> Gulag processing and kitchen</summary>

image

</details>

<details>
<summary> Gulag hydroponics and garden</summary>

image

</details>

This PR completely redoes the mining base and gulag area, giving it a new feel and adding a lot of character to both areas. The aim is as follows:
1. Make the gulag a place that feels like it's been lived in before. Give it character and a story.
2. Enable gulag prisoners to have something engaging to do. With the list of items, prisoners can choose to make something of their little corner of the universe or try and escape. There are no surefire ways to get out, so they will have to get creative with what they have. There are options to give security trouble, too, if that's more what they're looking to do.
3. Make the mining base feel like a mining base for people that are mining strategic assets. The public areas are nice, but the exclusive areas are nicer, and there's plenty of fanciful corporate shmoozing. I aimed to make sure that it still retained its industrial feel and had a layout that showed that it was conforming to the terrain it was on when it was built.
4. Give the gulag area some of the same exploration mechanics that regular mining has. Since gulag mining is boring, this may give people an incentive and purpose to their mining beyond getting points. It also sets the stage for exclusive gulag ruins in the near future if we want to go with that instead of static ruins.
5. Set the groundwork to give the gulag an opportunity to renew its resources or have mechanics to take advantage of prisoners. There are mining bins in the front of the gulag that a creative coder could use to set up mining bounties and raw mineral dropoffs for gulag prisoners to mine (for increased points? Exploitation of plausibly innocent crew? Who knows!)

Here's what I have left to do before it's ready:
- [x] Add in several areas, lore consoles, etc.
- [x] Add in all the wall objects including APCs
- [x] Add in atmospherics and other necessary machines
- [x] Review the area and items. Make sure I don't want to subtract or add anything else
- [x] Playtest on local to look for errors and to see if I can escape. Make pictures.
- [x] Finalize layout, items, and ruins
- [x] Testmerge to assess playability with players, ideally both on LRP and MRP servers

## Why It's Good For The Game

I feel like every punishment security can give to the crew that doesn't remove them from the round should still be something that they can actively participate and feel agency in; otherwise that option is, effectively, round removal. I wanted to make getting sent to the gulag something that you could turn into an interesting story rather than something that you dread doing or even worse something that makes you disconnect from the game. While it isn't perfect, now you can do some things other than mine and with the added ruins, there's progression to your story that can enable you to either resist security or make the gulag a really nice space. There are probably a few creative ways to escape from the gulag, but that is a part of the intentional flaws of the game.

Since I was remaking the gulag, I thought it would be a great opportunity to remake the mining station proper as well. We have a great mining station on icebox, and I wanted the mining station on lavaland to have just as much character. 

## Changelog

:cl:
add: Lavaland's Mining Station has received a serious facelift
add: Lots of new content for the gulag mining area, including miniature ruins, secrets, and more
/:cl:

### CODE REVIEW

**Comment by Elli-Skala** [comment]
That botany area seems to have of loot to steal 
Floral guns, BS beaker, bee's technically, a chem dispenser

**Comment by Son-of-Space** [comment]
> That botany area seems to have of loot to steal Floral guns, BS beaker, bee's technically, a chem dispenser

The dispenser only has mutagen and whatever sulfur thing botanists use.

**Comment by ATH1909** [comment]
> > That botany area seems to have of loot to steal Floral guns, BS beaker, bee's technically, a chem dispenser
> 
> The dispenser only has mutagen and whatever sulfur thing botanists use.

Expect botanists to still steal it (and the floral somatorays, ESPECIALLY the floral somatorays) anyway.

I'd recommend using the permabrig's botany setup as a better example of what botanical tools prisoners should be allowed to have.

**Comment by Son-of-Space** [comment]
> > > That botany area seems to have of loot to steal Floral guns, BS beaker, bee's technically, a chem dispenser
> > 
> > 
> > The dispenser only has mutagen and whatever sulfur thing botanists use.
> 
> Expect botanists to still steal it (and the floral somatorays, ESPECIALLY the floral somatorays) anyway.
> 
> I'd recommend using the permabrig's botany setup as a better example of what botanical tools prisoners should be allowed to have.

They've got the basics in the little one inside the gulag. The intent is that they have motivation to go out there, mine through the rock, and get that stuff. If it's a little too much I can definitely scale it back if someone gives a good argument.

I think if botanists break into security, fly down to a broken planet, get the ray, break OUT of what is supposedly the most secure place in security, and get back, they might be okay to have it.

**Comment by the-orange-cow** [comment]
> > > > That botany area seems to have of loot to steal Floral guns, BS beaker, bee's technically, a chem dispenser
> > > 
> > > 
> > > The dispenser only has mutagen and whatever sulfur thing botanists use.
> > 
> > 
> > Expect botanists to still steal it (and the floral somatorays, ESPECIALLY the floral somatorays) anyway.
> > I'd recommend using the permabrig's botany setup as a better example of what botanical tools prisoners should be allowed to have.
> 
> They've got the basics in the little one inside the gulag. The intent is that they have motivation to go out there, mine through the rock, and get that stuff. If it's a little too much I can definitely scale it back if someone gives a good argument.
> 
> I think if botanists break into security, fly down to a broken planet, get the ray, break OUT of what is supposedly the most secure place in security, and get back, they might be okay to have it.

Never underestimate the tiding power of a botanist.  I don't care if it was spawned directly north of the AI core round start, botany will make sure to obtain that gun within the first five minutes of the shift.

As it stands, they could just ask someone from Sec or the AI to let them reappropriate the gamer gear away from prisoners and towards the on station department that can actually get significant use out of it.

Don't get me wrong, as a botanist main I'd love the addition, but lets be real about where it will actually end up.

**Comment by ATH1909** [comment]
> I think if botanists break into security, fly down to a broken planet, get the ray, break OUT of what is supposedly the most secure place in security, and get back, they might be okay to have it.

"borg law 2 fetch me the somatoray from the gulag"

**Comment by Son-of-Space** [comment]
I think it's speculation until we see it. I also think ATHATH's example here is a bit disingenuous. A borg could be ordered to go traipsing around the space ruins for gear, but that doesn't mean they would (or could). They'd also have to be a mining borg, which means going and getting their modules reset. 99/100 times they have better things to do, a better order to follow, etc.

**Comment by optimumtact** [comment]
the gulag is round removal it's what it exists for, it's for that salty greytider you know will suicide the moment you sentence htem there.

**Comment by Son-of-Space** [comment]
> the gulag is round removal it's what it exists for, it's for that salty greytider you know will suicide the moment you sentence htem there.

We've got a nice isolation room for that.

**Comment by Redrover1760** [comment]
I love this PR. The mining station looks amazing now, and honesty I wouldn't mind hanging out down there or setting up something down there just to relax.

**Comment by Shadowflame909** [comment]
Its great. Only thing is that I cannot eat the glop in the kitchen because there are no bowls. (I assume thats what you use?) Please add a locker full of them. I'd say a vending machine but that has sharp knives and a prison wouldn't have that.

**Comment by timothymtorres** [comment]
This looks amazing! +1 from me

**Comment by TrashDoxx** [comment]
Good job 👍

**Comment by Son-of-Space** [comment]
image

**Comment by EOBGames** [comment]
^ The above DNM is so I can get round to reviewing this properly before merge, kindly poke me if it's been a few days without that and you want it removed because I'm too slow

**Comment by bruhlookatthisdood** [comment]
aww, thought this would be some new station project to plant tons of grass on lavaland and like make it habitable or something

**Review by Maurukas — COMMENTED** [review]
No ATs. No APC issues. Builds clean, server starts without errors or any other problems. Some really good mapping here. There are a lot of D4s scattered around the mining base and I stepped on one like eight times while reviewing.

Couple minor things:

<details>
<summary>Assuming these are supposed to be in the toilet, not on top of it. Doesn't appear that they can be stored there. </summary>

image

</details>
<details>
<summary> This weapon spawn is quite visible. Is this intended? </summary>

image
</details>

Other than that, this is a drastic improvement in the quality and detail of mapping for lavaland. I know Inept still has some concerns with this in terms of the prisoner areas. I see no technical issues, and the test merge on Manuel went extremely well.

**Comment by Son-of-Space** [comment]
It's unfortunate that I can't stuff the organs in the toilet but on top of it is fine. The weapon is intended to be there and if whomever wants to take it from them wants to go out into the wastes to grab it they are welcome to. There are other weapons spread around so that isn't the only one they have the potential to find. Thanks for the review! 👍

File: code/game/objects/effects/spawners/random/medical.dm [hunk 1]
```diff
@@ -39,6 +39,32 @@
...
+		/obj/item/organ/internal/ears/cat = 10,
+		/obj/item/organ/internal/eyes/fly = 10,
+		/obj/item/organ/internal/tongue/fly = 10,
+		/obj/item/organ/internal/tongue/tied = 8,
+		/obj/item/organ/external/tail/cat = 6,
+		/obj/item/organ/external/tail/lizard = 6,
+		/obj/item/organ/internal/liver/plasmaman = 5,
+		/obj/item/organ/internal/lungs/ashwalker = 5,
+		/obj/item/organ/internal/eyes/night_vision/mushroom = 3,
+		/obj/item/organ/internal/heart/gland/ventcrawling = 3,
```
Comment (EOBGames) [inline code/game/objects/effects/spawners/random/medical.dm:57]: let's not give the prisoners abductor organs, please

File: code/game/objects/structures/fluff.dm [hunk 2]
```diff
@@ -276,12 +276,24 @@
 	plane = FLOOR_PLANE
 	deconstructible = TRUE
 
+/obj/structure/fluff/tram_rail/lavaland
```
Comment (EOBGames) [inline code/game/objects/structures/fluff.dm:279]: does this need to be on a subtype? feels like lava resistance and fire resistance would be perfectly okay to put on the stationside tram rails too

**Review by EOBGames — CHANGES_REQUESTED** [review]
here's the code-side review, some stuff is going to need changed here

**Comment by EOBGames** [comment]
Now for the map side stuff, starting with the less controversial stuff (i.e. the mining base itself, where my concerns are mostly aesthetic) and then moving on to the gulag, where there are going to need to be a lot of changes.

So, the mining base:
image
These pamphlets have nothing written on them- seems a bit pointless to include, honestly. I'd suggest making something new, or just removing them entirely.

image
Making this the only public egress point from the mining base is a disaster waiting to happen- switch this up to a 2 wide catwalk and preferably add railings, especially since ore boxes are expected to be in play on this bit of the station.

image
These directional windows really don't need to be there- get rid of them, it'll make the room feel less confined _and_ reduce the number of unnecessary directional windows, making it a win-win.

image
Pick one window style and stick to it throughout the design- preferably the fulltiles.

image
This would look better with the windoor facing right, so you don't have to clamber over the toilet to take a shower.

image
Another directional window thing- I think this would be better as fulltiles.

Now, on to the gulag, and I'm sorry, but a lot needs changed.
image
In accordance with the prison kitchen equipment decree, this oven needs to go. The benches could also probably stand to be end pieces so they don't just sorta abruptly end. Also, you should include some of the tail pins (/obj/item/tail_pin) otherwise the poster is a bit useless.

image
let's not

image
I know this is sealed in, but giving prisoners 2/2 of the mandatory door hacking items feels like a misstep. I know we discussed creative escape methods on discord, but this allows way too much distilling down to a science.

image
Same with this multitool. Get rid of them, please.

I'm gonna just have to address the elephant in the room here- there's a lot of guaranteed stuff here that's always gonna be in set positions, and I don't believe turning the gulag into a "rush the gamer loot rooms" is a good idea. In the case of the botany equipment available, it's straight up better than what botany itself gets roundstart. I could go through and list it out, but to be honest I'd be listing the majority of the loot from the "extra ruins" on the islands. They're all going to need to be either thoroughly revamped, turned into non-guaranteed spawns (in the case of the less egregious stuff, like the bow and arrow), or removed entirely.

Since I know this is going to need extra attention from my colleagues, I'm inviting @san7890 and @Maurukas to also leave their thoughts on the loot available and @MrMelbert as an expert on botany to share their thoughts on the botany suite in particular. I'm willing to debate somewhat on the gulag contents, but in the PR's current state, it won't be going in.

**Comment by Son-of-Space** [comment]
The pamphlets have a general description instead of info so they get the gist of what's going on. If someone wants to add something lore-related to the info later that's fine but just something on examine is okay for now.

I want the public exit to be precarious, as if it was added on as an afterthought. Having mining station access gives them access to the NT approved way out.

The directional windows in that picture are there because it's a smoking lounge. I don't want to remove them.

I can't stick to full dir windows because the trash spawner is broken and is spawning trash underneath the windows in both that area and the maints by the bar area. Once that bug is fixed I will swap them out for the full dir windows, which are my preference because they give the base an industrial feel.

Kitchen: What do I give them to cook the meats and such that they would get from the ruin up north? I can make the other changes.

Flamethrower: They can't access it without tools and flamethrowers are very very easy to make on station. I want to keep it.

What do you propose as an alternative to giving them tools to allow them to sabotage things? Putting them behind a random spawner like the other permas do?

The multitool is solely there to allow them to break open abandoned crates. Would you rather me have locked crates that spawn a similar item that they can drag over lava?

**Comment by Son-of-Space** [comment]
I should also state that the screwdriver and wirecutter were not a misstep and were very deliberate. I wouldn't add something like that without fully understanding what I was doing. Since the gulag is completely isolated from the mining station now, they only way they could use those items to escape is by outside interference. They don't have insulated gloves either so if they want to hack doors, they need to be very deliberate about it.

**Comment by Maurukas** [comment]
I clearly need to look at the gulag more carefully as I missed most of this stuff.

**Comment by Son-of-Space** [comment]
I don't feel like there's a strong solution to maintain the creative premise behind this overhaul while also compromising with EOB, so I will be closing this. Please do not reopen it or attempt to add it in modified at a later date without discussing with me.

**Comment by optimumtact** [comment]
@Son-of-Space as soon as you started mapping it in the tgstation repository you no longer have the right to control distribution.

You are not owed that and people may choose for themselves what to do.

**Comment by TrashDoxx** [comment]
Based.

**Comment by Son-of-Space** [comment]
@optimumtact Let me make myself clear. I'm not speaking about controlling distribution. I'm speaking about giving me a basic level of respect as a long-time contributor who enjoys making this game better to allow me to decide if my work isn't where I want it to be to go on live and to involve me in that piece of the decision-making process. That's all. If you or someone else put up a copy of this tomorrow and it got merged, then that's certainly your decision, but I would be very disappointed.

**Comment by Son-of-Space** [comment]
I've decided that I would rather have these changes in the game now and have the opportunity to work on devices that will allow me to satisfy the vision for it while compromising with EOB at a later date. I'll be re-opening this and will be committing the requested changes shortly.

**Comment by Son-of-Space** [comment]
Requested changes have been made. Here's what I kept from the list:
1. The directional windows in the smokers' lounge.
2. A botany chem dispenser in the seed vault.
3. A condimaster in the seed vault
4. A singular pneumatic airlock seal swapped with the screwdriver and wirecutters
5. 30 plasma in the seed vault
6. The wired rod and cable coil
7. Practically everything in the spider ruin. Some of the decorative tile spawners were moved to the other room because it became barren.
8. The red crowbar that can't be retrieved without getting four tile stacks of lava burning

**Comment by Son-of-Space** [comment]
@MrMelbert @san7890 since EOB originally requested your input before I closed, I wanted to ping you again to let you know that this has been re-opened.

**Comment by Redrover1760** [comment]
> Kitchen: What do I give them to cook the meats and such that they would get from the ruin up north? I can make the other changes.

You can cook goliath meat with lava, perhaps you could make a way to make logs to make a campfire, perhaps?

**Comment by Elli-Skala** [comment]
> > Kitchen: What do I give them to cook the meats and such that they would get from the ruin up north? I can make the other changes.
> 
> You can cook goliath meat with lava, perhaps you could make a way to get logs or towercap mushrooms to make a campfire, perhaps? Gulag/perma prisoners making themselves a campfire to cook with sounds like a much more interesting method of cooking than being provided an oven. You can make the thing that goes on the campfire with rods so that's not hard.
> 
> Also you can already get towercap mushrooms by having stuff overgrown with weeds no? I don't know botany sorry.
> 
> All you'd need is a pr to allow a torch or something to be lit by lava that can light campfires or add a random cheap lighter that probably already exists somewhere.

A torch can be made from tower caps and dried leafy mushrooms

**Comment by EOBGames** [comment]
Alright, to start everything off, I'd like to offer an apology. For leaving this so long, for reviewing and dashing, for the lack of communication, for everything. I'm taking responsibility for all of that- it wasn't fair for me to be so uncommunicative, to attempt to hide behind the rest of my team, and to leave you high and dry for this long. I fully understand your frustrations about the way I've handled this, and all I can offer is apologies, and the promise that I will strive to do better in future. I also hope that you'll continue to work with us, although I would understand if this was your breaking point.

With that said, I am sorry to say that, of the two competing PRs for lavaland base reworks, we as a team prefer #67796. This is the result of a fair amount of discussion over the past 2 weeks, which I regret to say has mostly been behind closed doors. We've selected it for a number of factors- speaking for myself, I generally prefer the more island-based structure making good use of the planetary environment and the aesthetic choices are more to my preferences.

This PR is obviously not without merits. It has your characteristic eye for design and detail. It attempts to solve a problem that you've put forward, namely making gulag life a wider and more enjoyable experience. This is part of why the deliberation has taken so long- we've genuinely had to debate the strong points of each PR and which we would prefer to move ahead with.

To offer advice as to where to go from here, I would suggest that the issue of the gulag is not something that can reasonably be solved by mapping. I know we discussed this at some point on discord in the past month, and I'd reiterate that looking at a code solution is probably a better way to tackle the issues at hand.

**Pull request closed without merging**